sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.getInProgressApplicationDetails.e2e-spec.ts

```diff
@@ -0,0 +1,409 @@
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        id: currentApplication.id,
+        applicationStatus: currentApplication.applicationStatus,
+        pirStatus: currentApplication.pirStatus,
+        studentIncomeVerificationStatus: SuccessWaitingStatus.Success,
+        outstandingAssessmentStatus: SuccessWaitingStatus.Success,
+      });
+
+    // Change the Submitted and In-progress assessments to Completed to allow other test cases.
+    currentApplication.currentAssessment.studentAssessmentStatus =
+      StudentAssessmentStatus.Completed;
+    futureAssessment.studentAssessmentStatus =
+      StudentAssessmentStatus.Completed;
+    await db.studentAssessment.save([
+      currentApplication.currentAssessment,
+      futureAssessment,
+    ]);
+  });
+
+  it("Should get application in-progress details when there is an in-progress student assessment for an offering with earlier study start date.", async () => {
+    // Arrange
+
+    const currentApplication = await saveFakeApplication(db.dataSource, {
+      student,
+    });
+
+    // Create a previous application with an offering whose study start date and end date are much earlier
+    // than the current offering ones.
+    const previousApplication = await saveFakeApplication(db.dataSource, {
+      student,
+    });
+    const previousOffering = await db.educationProgramOffering.findOne({
+      select: {
+        id: true,
+        studyStartDate: true,
+        studyEndDate: true,
+      },
+      where: {
+        id: previousApplication.currentAssessment.offeringId,
+      },
+    });
+    previousOffering.studyStartDate = getISODateOnlyString(addDays(-60));
+    previousOffering.studyEndDate = getISODateOnlyString(addDays(-30));
+    await db.educationProgramOffering.save(previousOffering);
+
+    // Create an in-progress student assessment for the previous offering.
+    const previousAssessment = createFakeStudentAssessment(
+      {
+        auditUser: previousApplication.student.user,
+        application: previousApplication,
+        offering: previousOffering,
+      },
+      {
+        initialValue: {
+          calculationStartDate: addDays(-10),
+          studentAssessmentStatus: StudentAssessmentStatus.InProgress,
+        },
+      },
+    );
+    await db.studentAssessment.save(previousAssessment);
+
+    // Create CRA income verifications for student.
+    const studentCRAIncomeVerification = createFakeCRAIncomeVerification(
+      {
+        application: currentApplication,
+      },
+      { initialValues: { dateReceived: new Date() } },
+    );
+    await db.craIncomeVerification.save([studentCRAIncomeVerification]);
+
+    const endpoint = `/students/application/${currentApplication.id}/in-progress`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
```

Comment:
Please remove the income verification here. It makes the test case more complex to focus on test case specifics.